habitat/datasets/rearrange/sim_utilities.py

```diff
@@ -0,0 +1,567 @@
+                * distance
+                + look_at
+            )
+            return self.save_observation(
+                prefix="peek_" + obj.handle,
+                look_at=look_at,
+                look_from=look_from,
+            )
+        else:
+            # collect axis observations
+            axis_obs: List[Any] = []
+            for axis in range(6):
+                axis_vec = mn.Vector3()
+                axis_vec[axis % 3] = 1 if axis // 3 == 0 else -1
+                look_from = (
+                    obj_abs_transform.transform_vector(axis_vec).normalized()
+                    * distance
+                    + look_at
+                )
+                self.get_observation(look_at, look_from, axis_obs)
+            # stitch images together
+            stitched_image = None
+            from PIL import Image
+
+            from habitat_sim.utils import viz_utils as vut
+
+            for ix, obs in enumerate(axis_obs):
+                image = vut.observation_to_image(obs["rgb"], "color")
+                if stitched_image is None:
+                    stitched_image = Image.new(
+                        image.mode, (image.size[0] * 3, image.size[1] * 2)
+                    )
+                location = (
+                    image.size[0] * (ix % 3),
+                    image.size[1] * (0 if ix // 3 == 0 else 1),
+                )
+                stitched_image.paste(image, location)
+            stitched_image.show()
+        return ""
+
+    def make_debug_video(
+        self,
+        output_path: Optional[str] = None,
+        prefix: str = "",
+        fps: int = 4,
+        obs_cache: Optional[List[Any]] = None,
+    ) -> None:
+        """
+        Produce a video from a set of debug observations.
+        """
+        if output_path is None:
+            output_path = self.output_path
+
+        check_make_dir(output_path)
+
+        # get a timestamp tag with current date and time for video name
+        from datetime import datetime
+
+        date_time = datetime.now().strftime("%m_%d_%Y_%H%M%S")
+
+        if obs_cache is None:
+            obs_cache = self._debug_obs
+
+        from habitat_sim.utils import viz_utils as vut
+
+        file_path = output_path + prefix + date_time
+        print(f"DebugVisualizer: Saving debug video to {file_path}")
+        vut.make_video(
+            obs_cache, self.default_sensor_uuid, "color", file_path, fps=fps
+        )
+
+
+def check_make_dir(output_path: str) -> None:
+    """
+    Check for the existence of the provided path and create it if not found.
+    """
+    # if output directory doesn't exist, create it
+    if not osp.exists(output_path):
+        import os
+
+        try:
+            os.makedirs(output_path)
+        except OSError:
+            print(
+                f"DebugVisualizer: Aborting 'make_debug_video': Failed to create the output directory: {output_path}"
```

Comment:
what is `make_debug_video` ? I do not think we should assume that check_make_dir is always called by make_debug_video. Same with DebugVisualizer